src/cmakeTools.ts

```diff
@@ -646,6 +647,7 @@ export class CMakeTools implements api.CMakeToolsAPI {
                                 // doesn't need to fire the settings change event (which would trigger unnecessarily
                                 // another immediate configure, which will be blocked anyway).
                                 config.updatePartial({ sourceDirectory: relPath }, false);
+                                telemetryProperties["missingCMakeListsUserAction"] = "updateSourceDirectory";
```

Comment:
Do we also need to add something for `if (selectedFile)` (line 642) and `if (config)` (line 645) are not entered?

Reply:
I think having the else case is a good idea for case of invalid SourceDirectory, to ensure all cases are accounted for